include/aspect/simulator_signals.h

```diff
@@ -203,6 +203,18 @@ namespace aspect
      */
     boost::signals2::signal<void (typename parallel::distributed::Triangulation<dim> &)>  post_resume_load_user_data;
 
+    /**
+     * This signal is called whenever the pressure scaling is computed, see
+     * Simulator::compute_pressure_scaling_factor(), and allows inspection
+     * and/or modification of the computed factor.
+     *
+     * The argument @p pressure_scaling contains the computed pressure scaling (the ratio
+     * of the reference viscosity @p reference_viscosity computed by averaging
+     * the viscosity in the domain and the length scale @p length_scale reported
+     * by the geometry model) and can be used to modify the scaling factor.
+     */
+    boost::signals2::signal<void (double &pressure_scaling, const double reference_viscosity, const double length_scale)>  modify_pressure_scaling;
```

Comment:
Did you look into making this:
```suggestion
    boost::signals2::signal<double (const double pressure_scaling, const double reference_viscosity, const double length_scale)>  modify_pressure_scaling;
```

and then below do:
```
if (signals.modify_pressure_scaling.empty() == false)
  result = signals.modify_pressure_scaling(result, reference_viscosity, length_scale);
```
As far as I understand the boost documentation if there are several connected slots the signal will by default return the return value of the last slot function.

Reply:
I am in favor of @gassmoeller 's style.